typescript/web/src/components/app-lifecycle-manager/welcome-modal/welcome-modal.tsx

```diff
@@ -1,185 +1,357 @@
-import { useState, useEffect, useCallback, useRef } from "react";
+import {
+  // useRef,
+  useEffect,
+  useCallback,
+  useState,
+} from "react";
+import { ApolloClient, gql, useApolloClient } from "@apollo/client";
+import { useCookies } from "react-cookie";
+import { Modal, ModalOverlay } from "@chakra-ui/react";
 
+import { QueryParamConfig, StringParam, useQueryParam } from "use-query-params";
+import type { Dataset as DatasetType } from "@labelflow/graphql-types";
+import { useRouter, NextRouter } from "next/router";
+import { useErrorHandler } from "react-error-boundary";
+import { browser } from "../../../utils/detect-scope";
+import { BrowserWarning } from "./steps/browser-warning";
+import { BrowserError } from "./steps/browser-error";
+import { Welcome } from "./steps/welcome";
+import { Loading } from "./steps/loading";
 import {
-  chakra,
-  Modal,
-  ModalOverlay,
-  ModalContent,
-  ModalFooter,
-  VStack,
-  HStack,
-  Button,
-  Center,
-  Text,
-  Heading,
-  ModalBody,
-  ModalHeader,
-  useColorModeValue as mode,
-} from "@chakra-ui/react";
-import { RiGithubFill } from "react-icons/ri";
-import { StringParam, useQueryParam } from "use-query-params";
-
-import { Logo } from "../../logo";
-
-const GithubIcon = chakra(RiGithubFill);
+  checkServiceWorkerReady,
+  messageNoWindow,
+} from "../../../utils/check-service-worker";
 
-export const WelcomeModal = ({
+const tutorialDatasetFirstImageUrl =
+  "/local/datasets/tutorial-dataset/images/2bbbf664-5810-4760-a10f-841de2f35510";
+
+type WelcomeModalParam =
+  | undefined // Default: Stays close if user already onboarded, else open asap
+  | "open" // Force it to open asap
+  | "closed"; // Force it to be closed and never open
+
+export const getDatasetsQuery = gql`
+  query getDatasets {
+    datasets {
+      id
+      name
+    }
+  }
+`;
+
+export const createDemoDatasetQuery = gql`
+  mutation createDemoDataset {
+    createDemoDataset {
+      id
+      name
+      images(first: 1) {
+        id
+        url
+      }
+      imagesAggregates {
+        totalCount
+      }
+      labelsAggregates {
+        totalCount
+      }
+      labelClassesAggregates {
+        totalCount
+      }
+    }
```

Comment:
No

Reply:
But it's more as a safety feature. If this query fails then I prefer to catch it in the modal, and tell the guy there's a problem with his browser, instead of letting him go and let him have an error later on

Reply:
Ah ok! this makes sense.